llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -13479,6 +13501,56 @@ InstructionCost BoUpSLP::getGatherCost(ArrayRef<Value *> VL, bool ForPoisonSrc,
           Cost += TTI->getShuffleCost(
               TTI::SK_InsertSubvector, VecTy, std::nullopt, CostKind,
               I * ScalarTyNumElements, cast<FixedVectorType>(ScalarTy));
+    } else if (allConstant(VL)) {
+      auto IsAllowedScalarTy = [&](const Type *T) {
+        return T->isFloatTy() || T->isDoubleTy() || T->isIntegerTy();
+      };
```

Comment:
It should just call TTI getConstantMaterializationCost, nothing else

Reply:
done